src/main/java/org/usfirst/frc/team4099/robot/subsystems/Arm.kt

```diff
@@ -0,0 +1,170 @@
+import com.ctre.phoenix.motorcontrol.FeedbackDevice
+import com.ctre.phoenix.motorcontrol.can.TalonSRX
+
+
+/**
+ * @author Team 4099
+ *
+ * This class is the constructor for the Arm subsystem
+ *
+ * @constructor Creates the Arm subsystem
+ *
+ */
+class Arm private constructor() : Subsystem {
+
+    private val masterSRX = TalonSRX(Constants.Arm.MASTER_SRX_ID)
+    private val slaveSRX1 = TalonSRX(Constants.Arm.SLAVE_SRX_1_ID)
+    private val slaveSRX2 = TalonSRX(Constants.Arm.SLAVE_SRX_2_ID)
+    private val slaveSRX3 = TalonSRX(Constants.Arm.SLAVE_SRX_3_ID)
+
+    private val brake = DoubleSolenoid(Constants.Arm.BRAKE_FORWARD_CHANNEL, Constants.Arm.BRAKE_REVERSE_CHANNEL)
+
+
+    var movementState = MovementState.STATIONARY
+    private var armPower = 0.0
+    private var armAngle = 0.0
+    private var armBaseAngle = 0.0
+
+    private var stationaryTime = 0
+
+
+    var armState = ArmState.EXCHANGE
+
+
+    var useVelocityControl : Boolean = false
+
+    enum class MovementState {
+        UP, STATIONARY, DOWN, MOVING_TO_EXCHANGE, MOVING_TO_TOP, MOVING_TO_BOTTOM
+    }
+
+    enum class ArmState(val targetPos: Double) {
+        LOW(-7* Math.PI / 6), EXCHANGE(0.0), HIGH(7 * Math.PI / 6), VELOCITY_CONTROL(Double.NaN)
+    }
+
+
+
+
+    init {
+        masterSRX.set(ControlMode.MotionMagic, 0.0)
+        slaveSRX1.set(ControlMode.Follower, Constants.Arm.MASTER_SRX_ID.toDouble())
+        slaveSRX2.set(ControlMode.Follower, Constants.Arm.MASTER_SRX_ID.toDouble())
+        slaveSRX3.set(ControlMode.Follower, Constants.Arm.MASTER_SRX_ID.toDouble())
+        masterSRX.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative, 0, 10)
+        masterSRX.setSensorPhase(false)
+        masterSRX.configNominalOutputForward(+0.0, 0)
+        masterSRX.configPeakOutputForward(+1.0, 0)
+        masterSRX.selectProfileSlot(0, 0)
+        masterSRX.config_kF(0, 0.0, 0)
+        masterSRX.config_kP(0, 0.0, 0)
+        masterSRX.config_kI(0, 0.0, 0)
+        masterSRX.config_kD(0, 0.0, 0)
+        armBaseAngle = ArmConversion.pulsesToRadians(masterSRX.sensorCollection.pulseWidthPosition)
+
+    }
+
+
+    /**
+     * Changes power to the arm
+     *
+     * @param power A double for the power going to the arm
+     */
+    fun setArmVelocity(power: Double) {
+        armState = ArmState.VELOCITY_CONTROL
+        masterSRX.set(ControlMode.Velocity, Math.abs(power))
+        armPower = power
+    }
+
+    private fun setArmPosition(position: Double) {
+        armAngle = position
+        masterSRX.set(ControlMode.MotionMagic, ArmConversion.radiansToPulses(position).toDouble())
+    }
+
+    /**
+     * Outputs arm subsystem
+     *
+     */
+    override fun outputToSmartDashboard() {
+        SmartDashboard.putNumber("armPower", armPower)
+    }
+
+    @Synchronized override fun stop() {
+        movementState = MovementState.STATIONARY
+        setArmVelocity(0.0)
+        brake.set(DoubleSolenoid.Value.kReverse)
```

Comment:
We might have to switch all the `kReverse` and `kForward` settings later, but that's not that big of a deal.